Crash occurs when attempting to configure the Farming HUB

### Minecraft Version

1.20.4

### Skyblocker Version

skyblocker-1.19-beta.1+1.20.4.jar

### Reproduction Steps

1. Download [Skyblocker](https://modrinth.com/mod/skyblocker-liap/version/1.19-beta.1+1.20.4), [Mod Menu](https://modrinth.com/mod/modmenu/version/9.0.0), [Fabric API](https://modrinth.com/mod/fabric-api/version/0.96.11+1.20.4).
2. Launch the game and go to mods.
3. Configure Skyblocker.
4. Go to `Locations`
5. Scroll down and click `Farming HUB Config`
6. crash :(

### Crash Report file and latest.log

```shell
Mods:
  [✔] fabric-api-0.96.11 1.20.4
  [✔] modmenu-9.0.0
  [✔] skyblocker-1.19-beta.1 1.20.4

---- Minecraft Crash Report ----
// This doesn't make any sense!

Time: 2024-03-22 07:03:53
Description: mouseClicked event handler

java.lang.ExceptionInInitializerError
	at de.hysky.skyblocker.skyblock.garden.FarmingHudConfigScreen.<init>(FarmingHudConfigScreen.java:14)
	at de.hysky.skyblocker.config.categories.LocationsCategory.lambda$create$27(LocationsCategory.java:153)
	at dev.isxander.yacl3.gui.controllers.ActionController$ActionControllerElement.executeAction(ActionController.java:79)
	at dev.isxander.yacl3.gui.controllers.ActionController$ActionControllerElement.method_25402(ActionController.java:85)
	at dev.isxander.yacl3.gui.ElementListWidgetExt$Entry.method_25402(ElementListWidgetExt.java:153)
	at net.minecraft.client.gui.widget.EntryListWidget.mouseClicked(EntryListWidget.java:279)
	at dev.isxander.yacl3.gui.OptionListWidget.method_25402(OptionListWidget.java:149)
	at dev.isxander.yacl3.gui.tab.ListHolderWidget.method_25402(ListHolderWidget.java:56)
	at net.minecraft.client.gui.ParentElement.mouseClicked(ParentElement.java:38)
	at net.minecraft.client.Mouse.method_1611(Mouse.java:99)
	at net.minecraft.client.gui.screen.Screen.wrapScreenError(Screen.java:414)
	at net.minecraft.client.Mouse.onMouseButton(Mouse.java:99)
	at net.minecraft.client.Mouse.method_22686(Mouse.java:180)
	at net.minecraft.util.thread.ThreadExecutor.execute(ThreadExecutor.java:102)
	at net.minecraft.client.Mouse.method_22684(Mouse.java:180)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3509)
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:238)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1352)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient.java:888)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.class_746.method_6047()" because "this.client.field_1724" is null
	at de.hysky.skyblocker.skyblock.garden.FarmingHudWidget.updateContent(FarmingHudWidget.java:55)
	at de.hysky.skyblocker.skyblock.tabhud.widget.Widget.update(Widget.java:53)
	at de.hysky.skyblocker.skyblock.garden.FarmingHudWidget.<init>(FarmingHudWidget.java:49)
	at de.hysky.skyblocker.skyblock.garden.FarmingHudWidget.<clinit>(FarmingHudWidget.java:42)
	... 28 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at de.hysky.skyblocker.skyblock.garden.FarmingHudConfigScreen.<init>(FarmingHudConfigScreen.java:14)
	at de.hysky.skyblocker.config.categories.LocationsCategory.lambda$create$27(LocationsCategory.java:153)
	at dev.isxander.yacl3.gui.controllers.ActionController$ActionControllerElement.executeAction(ActionController.java:79)
	at dev.isxander.yacl3.gui.controllers.ActionController$ActionControllerElement.method_25402(ActionController.java:85)
	at dev.isxander.yacl3.gui.ElementListWidgetExt$Entry.method_25402(ElementListWidgetExt.java:153)
	at net.minecraft.client.gui.widget.EntryListWidget.mouseClicked(EntryListWidget.java:279)
	at dev.isxander.yacl3.gui.OptionListWidget.method_25402(OptionListWidget.java:149)
	at dev.isxander.yacl3.gui.tab.ListHolderWidget.method_25402(ListHolderWidget.java:56)
	at net.minecraft.client.gui.ParentElement.mouseClicked(ParentElement.java:38)
	at net.minecraft.client.Mouse.method_1611(Mouse.java:99)
	at net.minecraft.client.gui.screen.Screen.wrapScreenError(Screen.java:414)
	at net.minecraft.client.Mouse.onMouseButton(Mouse.java:99)
	at net.minecraft.client.Mouse.method_22686(Mouse.java:180)
	at net.minecraft.util.thread.ThreadExecutor.execute(ThreadExecutor.java:102)
	at net.minecraft.client.Mouse.method_22684(Mouse.java:180)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3509)

-- Affected screen --
Details:
	Screen name: dev.isxander.yacl3.gui.YACLScreen
Stacktrace:
	at net.minecraft.client.gui.screen.Screen.wrapScreenError(Screen.java:414)
	at net.minecraft.client.Mouse.onMouseButton(Mouse.java:99)
	at net.minecraft.client.Mouse.method_22686(Mouse.java:180)
	at net.minecraft.util.thread.ThreadExecutor.execute(ThreadExecutor.java:102)
	at net.minecraft.client.Mouse.method_22684(Mouse.java:180)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3509)
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:238)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1352)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient.java:888)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)

-- Uptime --
Details:
	JVM uptime: 45.576s
	Wall uptime: 22.784s
	High-res time: 19.739s
	Client ticks: 334 ticks / 16.700s
Stacktrace:
	at net.minecraft.client.MinecraftClient.addDetailsToCrashReport(MinecraftClient.java:2564)
	at net.minecraft.client.MinecraftClient.printCrashReport(MinecraftClient.java:995)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient.java:908)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
```


### Additional context

_No response_

--- a/src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java
+++ b/src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java
@@ -52,7 +52,7 @@ public class FarmingHudWidget extends Widget {
 
     @Override
     public void updateContent() {
-        ItemStack icon = client.player == null ? Ico.HOE: FARMING_TOOLS.getOrDefault(ItemUtils.getItemId(client.player.getMainHandStack()), Ico.HOE);
+        ItemStack icon = client.player == null ? Ico.HOE : FARMING_TOOLS.getOrDefault(ItemUtils.getItemId(client.player.getMainHandStack()), Ico.HOE);
         addSimpleIcoText(icon, "Counter: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format(FarmingHud.counter()));
         addSimpleIcoText(icon, "Crops/min: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format((int) FarmingHud.cropsPerMinute() / 100 * 100));
         addSimpleIcoText(icon, "Blocks/s: ", Formatting.YELLOW, Integer.toString(FarmingHud.blockBreaks()));
@@ -61,8 +61,8 @@ public class FarmingHudWidget extends Widget {
         addSimpleIcoText(Ico.LIME_DYE, "Farming XP/h: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format((int) FarmingHud.farmingXpPerHour()));
 
         Entity cameraEntity = client.getCameraEntity();
-        double yaw = cameraEntity == null ? 0.0d: cameraEntity.getYaw();
-        double pitch = cameraEntity == null ? 0.0d: cameraEntity.getPitch();
+        double yaw = cameraEntity == null ? 0.0d : cameraEntity.getYaw();
+        double pitch = cameraEntity == null ? 0.0d : cameraEntity.getPitch();
         addComponent(new PlainTextComponent(Text.literal("Yaw: " + String.format("%.3f", MathHelper.wrapDegrees(yaw))).formatted(Formatting.YELLOW)));
         addComponent(new PlainTextComponent(Text.literal("Pitch: " + String.format("%.3f", MathHelper.wrapDegrees(pitch))).formatted(Formatting.YELLOW)));
     }

--- a/src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java
+++ b/src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java
@@ -52,7 +52,7 @@ public class FarmingHudWidget extends Widget {
     @SuppressWarnings("DataFlowIssue")
     @Override
     public void updateContent() {
-        ItemStack icon = FARMING_TOOLS.getOrDefault(ItemUtils.getItemId(client.player.getMainHandStack()), Ico.HOE);
+        ItemStack icon = client.player == null ? Ico.HOE: FARMING_TOOLS.getOrDefault(ItemUtils.getItemId(client.player.getMainHandStack()), Ico.HOE);
         addSimpleIcoText(icon, "Counter: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format(FarmingHud.counter()));
         addSimpleIcoText(icon, "Crops/min: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format((int) FarmingHud.cropsPerMinute() / 100 * 100));
         addSimpleIcoText(icon, "Blocks/s: ", Formatting.YELLOW, Integer.toString(FarmingHud.blockBreaks()));

--- a/src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java
+++ b/src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java
@@ -7,6 +7,7 @@ import de.hysky.skyblocker.skyblock.tabhud.widget.component.PlainTextComponent;
 import de.hysky.skyblocker.skyblock.tabhud.widget.component.ProgressComponent;
 import de.hysky.skyblocker.utils.ItemUtils;
 import net.minecraft.client.MinecraftClient;
+import net.minecraft.entity.Entity;
 import net.minecraft.item.ItemStack;
 import net.minecraft.text.MutableText;
 import net.minecraft.text.Text;
@@ -49,18 +50,19 @@ public class FarmingHudWidget extends Widget {
         update();
     }
 
-    @SuppressWarnings("DataFlowIssue")
     @Override
     public void updateContent() {
         ItemStack icon = client.player == null ? Ico.HOE: FARMING_TOOLS.getOrDefault(ItemUtils.getItemId(client.player.getMainHandStack()), Ico.HOE);
         addSimpleIcoText(icon, "Counter: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format(FarmingHud.counter()));
         addSimpleIcoText(icon, "Crops/min: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format((int) FarmingHud.cropsPerMinute() / 100 * 100));
         addSimpleIcoText(icon, "Blocks/s: ", Formatting.YELLOW, Integer.toString(FarmingHud.blockBreaks()));
+        //noinspection DataFlowIssue
         addComponent(new ProgressComponent(Ico.LANTERN, Text.literal("Farming Level: "), FarmingHud.farmingXpPercentProgress(), Formatting.GOLD.getColorValue()));
         addSimpleIcoText(Ico.LIME_DYE, "Farming XP/h: ", Formatting.YELLOW, FarmingHud.NUMBER_FORMAT.format((int) FarmingHud.farmingXpPerHour()));
 
-        double yaw = client.getCameraEntity().getYaw();
-        double pitch = client.getCameraEntity().getPitch();
+        Entity cameraEntity = client.getCameraEntity();
+        double yaw = cameraEntity == null ? 0.0d: cameraEntity.getYaw();
+        double pitch = cameraEntity == null ? 0.0d: cameraEntity.getPitch();
         addComponent(new PlainTextComponent(Text.literal("Yaw: " + String.format("%.3f", MathHelper.wrapDegrees(yaw))).formatted(Formatting.YELLOW)));
         addComponent(new PlainTextComponent(Text.literal("Pitch: " + String.format("%.3f", MathHelper.wrapDegrees(pitch))).formatted(Formatting.YELLOW)));
     }

